rs/bitcoin/adapter/src/lib.rs

```diff
@@ -193,16 +193,31 @@ impl AdapterState {
 }
 
 /// Starts the gRPC server and the router for handling incoming requests.
-pub fn start_grpc_server_and_router(
-    config: &config::Config,
+pub fn start_server(
+    log: &ReplicaLogger,
     metrics_registry: &MetricsRegistry,
-    logger: ReplicaLogger,
-    adapter_state: AdapterState,
+    rt_handle: &tokio::runtime::Handle,
+    config: config::Config,
 ) {
+    let _enter = rt_handle.enter();
+
+    // Metrics server should only be started if we are managed by systemd and receive the
+    // metrics socket as FD(4).
+    // SAFETY: The process is managed by systemd and is configured to start with at metrics socket.
+    // Additionally this function is only called once here.
+    // Systemd Socket config: ic-https-outcalls-adapter.socket
+    // Systemd Service config: ic-https-outcalls-adapter.service
+    if config.incoming_source == IncomingSource::Systemd {
+        let stream = unsafe { incoming_from_nth_systemd_socket(2) };
```

Comment:
in the comment, you refer to FD(4)

Reply:
I believe the comment is not for the BTC adapter but for the socket that the adapter metrics use